turnitintooltwo_view.class.php

```diff
@@ -636,10 +636,18 @@ public function init_submission_inbox($cm, $turnitintooltwoassignment, $partdeta
                 // Check that nonsubmitter messages have been configured to be sent.
                 $messageoutputs = get_config('message');
                 $nonsubsemailpermitted = false;
-                foreach ($messageoutputs as $k => $v) {
-                    if (strpos($k, '_mod_turnitintooltwo_nonsubmitters_loggedin') !== false ) {
+
+                if ($CFG->branch >= 400) {
```

Comment:
This block can be extracted as a private function, which can simply be called and checked for the value returned. This would help with the readability of the code. Also creates a single point of call where this can be checked and called. 

If instead of adding it as a private method in this class, you could add it as a "static like" method in a Utility class, then you are getting something that can be called anywhere you might need it again. But for now I would say at least refactor this to abstract it in its own method. E.g.

```
private function permit_non_submitter_emails($messageoutputs) {

    if ($CFG->branch >= 400) {
        if (isset($messageoutputs->mod_turnitintooltwo_nonsubmitters_disable) && $messageoutputs->mod_turnitintooltwo_nonsubmitters_disable == "0") {
            return true;
        }
    } else {
        // Support for older versions.
        foreach ($messageoutputs as $k => $v) {
            if (strpos($k, '_mod_turnitintooltwo_nonsubmitters_loggedin') !== false) {
                return true;
                break;
            }
        }
        return false;
    }
}
```

Reply:
This will only ever be used once so yeah the latter way makes more sense.